lesson6-questions.txt

```diff
@@ -3,35 +3,61 @@ Questions for Lesson 6
 that describes the route.  How do you do that? What variable is created?
 
 
+When you specify a route in config/routes.rb, you need to include an HTTP verb, like GET, POST, PUT,
+or DELETE; a URL path that triggers the route, and a controller action that handles the request. You may specify
+a route with a parameter that you can use in your controller action. This parameter is extracted from
+the URL and made available in the params hash. For example, get '/users/:id', to: 'users#show', as: 'user',
+in which we see the HTTP verb 'get' and the URL pattern, '/users/:id'. This URL pattern has :id as a placeholder
+the captures the value from the URL and makes it available in the controller action as 'params[:id]'.
+So, the users will have a user_id, and the controller will be able to use that id captured from the URL. Then,
+to: 'users#show' shows that the request is handled by the show action in the user_controller.
 
-3. What does the Rails db:migrate command do?
 
+3. What does the Rails db:migrate command do?
 
+Migrations modify the database. This command will apply pending changes in order by looking at the timestamp to
+determine which should go first, it executes migration code that creates or alters the database, and updates the schema
+version so that it can keep track of which migrations have already been done.
 
 4. What file do you look at to see the current database schema?
-
+To see the current database schema, you can look at db/schema.rb. This is automatically updated whenever you run
+db:migrate.
 
 5. What command shows the routes?
 
-
+bin/rails routes
 6. How can you start the Rails console? How can you do CRUD operations from the console?
-
+You can start the rails console by typing: bin/rails console. Then, you can create new instances and define 
+the attributes of those instances. Afterward they can be saved, and you can view it by using a method that gets the
+data from the database.
 
 7. What statements are in erb files that make them different from html files?  What are the
 two kinds of special statements, and how are they marked in the erb file? Where are they executed?
 
+erb files contain embedded Ruby blocks that are executed on the server side to generate plain HTML which is sent to the
+browser. Some start with <%, and they are executed but do not result in anything added to the HTML. They are for conditional
+statements and loops. Other blocks start with <%=, and they do generate output.
 
 8. Consider this statement: @forum = Forum.new().  Why is Forum capitalized?  Why is @forum not
 capitalized? Why is the @ used?
 
+Forum is capitalized because it is calling a class method, making a new instance of the Forum class.
+@forum is the instance variable, so it is not capitalized. The instance variable can be accessed by any instance method
+in the class, and it is used to store data that is unique to each instance of a class.
```

Comment:
Good, this is correct. More specifically though, `Forum` is capitalized because `Forum` is the name of a class, class names are _constants_, and constants begin with a capital letter.